earthaccess/store.py

```diff
@@ -200,40 +199,49 @@ def get_s3fs_session(
         Returns:
             a s3fs file instance
         """
-        if self.auth is not None:
-            if not any([concept_id, daac, provider, endpoint]):
-                raise ValueError(
-                    "At least one of the concept_id, daac, provider or endpoint"
-                    "parameters must be specified. "
-                )
+        if self.auth is None:
+            raise ValueError(
+                "A valid Earthdata login instance is required to retrieve S3 credentials"
+            )
+        if not any([concept_id, daac, provider, endpoint]):
+            raise ValueError(
+                "At least one of the concept_id, daac, provider or endpoint"
+                "parameters must be specified. "
+            )
+
+        # Get existing S3 credentials if we already have them
+        location = (concept_id, daac, provider, endpoint)
+        need_new_creds = False
+        try:
+            dt_init, creds = self._s3_credentials[location]
+        except KeyError:
+            need_new_creds = True
+        else:
+            delta = datetime.datetime.now() - dt_init
+            if round(delta.seconds / 60, 2) > 55:
+                need_new_creds = True
+                self._s3_credentials.pop(location)
+
+        if need_new_creds:
+            # Don't have existing valid S3 credentials, so get new ones
+            now = datetime.datetime.now()
             if endpoint is not None:
-                s3_credentials = self.auth.get_s3_credentials(endpoint=endpoint)
+                creds = self.auth.get_s3_credentials(endpoint=endpoint)
```

Comment:
Weak nit: I kind of prefer longer variable names like `s3_credentials` :) What was the rationale for the name change?

Reply:
I also like more descriptive variable names when there's not much other context around, but when there is sufficient context, more succinct names tend to reduce the cognitive load on readers (at least in my personal opinion). 

In this case this method is all about S3, and this variable is coming from a method called `get_s3_credentials`, so my sense is there's enough context to know that `creds` are S3 credentials.

Reply:
Makes total sense to me, thanks for reasoning that out.